helpers/integration.go

```diff
@@ -180,63 +178,101 @@ func (s SortTaskRunsByStartTime) Less(i int, j int) bool {
-	var results []*AppStudioTestResult
-	var err error
-	// Check if the pipelineRun finished from the condition of status
-	if !HasPipelineRunFinished(pipelineRun) {
-		logger.Info(fmt.Sprintf("PipelineRun %s in namespace %s has not finished", pipelineRun.Name, pipelineRun.Namespace))
-		return false, nil
+func (ipro *IntegrationPipelineRunOutcome) GetOutcome() bool {
+	if !ipro.HasPipelineRunSucceeded() {
+		return false
 	}
+	for _, result := range ipro.results {
+		if result.Result != AppStudioTestOutputSuccess && result.Result != AppStudioTestOutputSkipped {
+			return false
+		}
+	}
+	return true
+}
+
+// LogResults writes tasks names with results into given logger, each task on separate line
+func (ipro *IntegrationPipelineRunOutcome) LogResults(logger logr.Logger) {
+	for k, v := range ipro.results {
+		logger.Info(fmt.Sprintf("Found task results for pipeline run %s", ipro.pipelineRunName),
+			"pipelineRun.Name", ipro.pipelineRunName, "task.Name", k, "task.Result", v)
+	}
+}
+
+// GetIntegrationPipelineRunOutcome returns the IntegrationPipelineRunOutcome which can be used for further inspection of
+// the results and general outcome
+// This function must be called on the finished pipeline
+func GetIntegrationPipelineRunOutcome(adapterClient client.Client, ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (*IntegrationPipelineRunOutcome, error) {
+
 	// Check if the pipelineRun failed from the conditions of status
 	if !HasPipelineRunSucceeded(pipelineRun) {
-		logger.Error(fmt.Errorf("PipelineRun %s in namespace %s failed for %s", pipelineRun.Name, pipelineRun.Namespace, GetPipelineRunFailedReason(pipelineRun)), "PipelineRun failed without test results of TaskRuns")
-		return false, nil
+		return &IntegrationPipelineRunOutcome{
+			pipelineRunSucceeded: false,
+			pipelineRunName:      pipelineRun.Name,
+			results:              map[string]*AppStudioTestResult{},
+		}, nil
 	}
 	// Check if the pipelineRun.Status contains the childReferences to TaskRuns
 	if !reflect.ValueOf(pipelineRun.Status.ChildReferences).IsZero() {
 		// If the pipelineRun.Status contains the childReferences, parse them in the new way by querying for TaskRuns
-		results, err = GetAppStudioTestResultsFromPipelineRunWithChildReferences(adapterClient, ctx, logger, pipelineRun)
+		results, err := GetAppStudioTestResultsFromPipelineRunWithChildReferences(adapterClient, ctx, pipelineRun)
 		if err != nil {
-			return false, fmt.Errorf("error while getting test results from pipelineRun %s: %w", pipelineRun.Name, err)
-		}
-	}
-
-	for _, result := range results {
-		if result.Result != AppStudioTestOutputSuccess && result.Result != AppStudioTestOutputSkipped {
-			return false, nil
+			return nil, fmt.Errorf("error while getting test results from pipelineRun %s: %w", pipelineRun.Name, err)
 		}
+		return &IntegrationPipelineRunOutcome{
+			pipelineRunSucceeded: true,
```

Comment:
Why we set `pipelineRunSucceeded` to `true` here? I think the result of taskRun might be `Failed`

Reply:
PLR itself succeeded, it doesn't record task results